Il2CppInspector.Common/Outputs/ScriptResources/shared-main.py

```diff
@@ -96,9 +96,16 @@ def ProcessJSON(jsonData):
 		DefineILMethodInfo(d)
 
 	# Function boundaries
-	print('Processing function boundaries')
-	for d in jsonData['functionAddresses']:
-		MakeFunction(int(d, 0))
+	functionAddresses = jsonData['functionAddresses']
+	count = len(functionAddresses)
+	for i in range(count):
+		addrStart = int(functionAddresses[i],0)
+		if addrStart == 0:
+			continue
+		addrNext = None
+		if i != count -1:
+			addrNext = int(functionAddresses[i+1],0)
+		MakeFunction(addrStart,None,addrNext)
```

Comment:
This implementation assumes that the function addresses are sorted in ascending order. Although I do populate the C# Dictionary like that, the CLR makes no guarantee that the elements will be retrieved in the order they were added to the collection. So you either need to sort them in the code here, or at https://github.com/djkaty/Il2CppInspector/blob/835636d99e0897a9401f9398e4b5a5bd94fd27b8/Il2CppInspector.Common/Outputs/JSONMetadata.cs#L142

Reply:
yes,this could simple add `functionAddresses.sort()` since they are all address string,sort order will same as int